core/loadpoint_effective.go

```diff
@@ -59,16 +78,12 @@ func (lp *Loadpoint) nextVehiclePlan() (time.Time, int, int) {
 				continue
 			}
 
-			plans = append(plans, plan{Id: index + 2, Soc: rp.Soc, Time: time})
+			plans = append(plans, plan{Id: index + 2, Soc: rp.Soc, End: time})
 		}
 
-		// sort plans by time
-		sort.Slice(plans, func(i, j int) bool {
-			return plans[i].Time.Before(plans[j].Time)
-		})
-
-		if len(plans) > 0 {
-			return plans[0].Time, plans[0].Soc, plans[0].Id
+		// calculate earliest required plan start
+		if plan := lp.nextActivePlan(lp.EffectiveMaxPower(), plans); plan != nil {
```

Comment:
```suggestion
		if plan := lp.nextActivePlan(plans); plan != nil {
```

Das `lp.EffectiveMaxPower()` muss eigentlich nicht reingereicht werden. Das kann nextActivePlan sich ja selbst raussuchen. Für den Unittest ist das praktisch, aber wir rufen das ja nie mit unterschiedlichen Powers auf.

Reply:
@naltatis der Grund war die Testbarkeit :(